### PR TITLE
Refactoring testing related python packages and SWIG wrapper bug fix

### PR DESCRIPTION
**Pull request by michaeltryby**
This PR contains some refactoring of testing related python packages and SWIG wrapper bug fix. 

Eliminated `epanet-reader` package. Removed `numpy` dependency from `epanet-output`. Fixed reference counting bug in SWIG wrapper. Added error checking to `run_nrtest.sh`. Added OWA's fork of the `nrtest` package to the requirements file.

### CODE REVIEW

**Comment by eladsal** [comment]
Apart for the PI fix there is no change to EPANET core. I'm fine with merging.

**Comment by michaeltryby** [comment]
@eladsal Thanks for the review!  @samhatchett any thoughts on this before I merge it?

File: tools/epanet-output/test/test_epanet_output.cpp [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+ *   Author: Michael E. Tryby
+ *           US EPA - ORD/NRMRL
+ *
+ *   Unit testing for EPANET Output API using google test.
+*/
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string>
+#include "gtest/gtest.h"
```
Comment (samhatchett) [inline tools/epanet-output/test/test_epanet_output.cpp]: Google Test? were others considered?

File: tools/epanet-output/test/test_epanet_output.cpp [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+#include "../src/epanet_output.h"
+
+#define PROJECT_HOME "C:/Users/mtryby/Workspace/GitRepo/michaeltryby/epanet/"
```
Comment (samhatchett) [inline tools/epanet-output/test/test_epanet_output.cpp]: should remove hard-coded local paths specific to a machine

File: tools/epanet-output/test/test_epanet_output.cpp (hunk 1 above)
Comment (michaeltryby) [reply tools/epanet-output/test/test_epanet_output.cpp]: I am just getting into unit testing. I immediately appreciated the value in running and rerunning tests as part of the development process. I found google test easy to setup using msys2 and easy to use. I am not heavily invested in `gtest` at this point and intend to try other testing frameworks prior to getting "religious" about it.

File: tools/epanet-output/test/test_epanet_output.cpp (hunk 2 above)
Comment (michaeltryby) [reply tools/epanet-output/test/test_epanet_output.cpp]: I understand this is not ideal. The test needs to know the location of the binary file to run. Is there a better way to accomplish this?

Comment (samhatchett) [reply tools/epanet-output/test/test_epanet_output.cpp]: as a command-line argument passed in by Travis?

**Comment by michaeltryby** [comment]
@samhatchett The message "Unexpected error occurred during diff" from `nrtest` means that a difference was found between the binary files. I agree that this message is somewhat ambiguous.

**Comment by samhatchett** [comment]
So then what is the difference between "Example 1: example1.rpt: diff failed" and "Unexpected error occurred during diff"

**Comment by michaeltryby** [comment]
> So then what is the difference between "Example 1: example1.rpt: diff failed" and "Unexpected error occurred during diff"

The current comparison routines are designed to fail fast. So as soon as a difference is detected they return a false value. I believe that the report files are being compared before the binary files. The report file test is very strict - requiring the files to match exactly. 

So what is happening in Examples 1 and 3 is that a difference was detected in the report file so a diff of the binary files never occurred. With Example 2, the report file diff passed and the diff of the binary files is failing.

**Comment by eladsal** [comment]
@michaeltryby where are the the test networks located?

**Comment by michaeltryby** [comment]
@eladsal Test networks are located [here](https://github.com/OpenWaterAnalytics/EPANET/tree/feature-wrapper/tests/epanet-nrtestsuite) in the tests directory.

**Comment by michaeltryby** [comment]
@eladsal and @samhatchett I propose we merge this PR as is and I will take care of the test hard wiring issue when this gets moved over to `epanet-python`. Sound good?

**Comment by eladsal** [comment]
I'm fine with it.

**Comment by michaeltryby** [comment]
Thanks guys!

**Pull request merged**